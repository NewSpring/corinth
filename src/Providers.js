import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { AuthProvider } from '@apollosproject/ui-auth';
import { AnalyticsProvider } from '@apollosproject/ui-analytics';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { NotificationsProvider } from '@apollosproject/ui-notifications';
import { LiveProvider } from '@apollosproject/ui-connected';

import NavigationService from './NavigationService';
import ClientProvider from './client';
import ExternalLinkProvider from './linking/Provider';
import track from './analytics';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider {...props}>
    <NotificationsProvider
      oneSignalKey={ApollosConfig.ONE_SIGNAL_KEY}
      navigate={NavigationService.navigate}
    >
      <AuthProvider
        navigateToAuth={() => NavigationService.navigate('Auth')}
        navigate={NavigationService.navigate}
        closeAuth={() => NavigationService.navigate('Onboarding')}
      >
        <ExternalLinkProvider navigate={NavigationService.navigate}>
          <MediaPlayerProvider>
            <AnalyticsProvider trackFunctions={[track]}>
              <LiveProvider>
                <Providers
                  themeInput={customTheme}
                  iconInput={customIcons}
                  {...props}
                />
              </LiveProvider>
            </AnalyticsProvider>
          </MediaPlayerProvider>
        </ExternalLinkProvider>
      </AuthProvider>
    </NotificationsProvider>
  </ClientProvider>
);

export default AppProviders;
