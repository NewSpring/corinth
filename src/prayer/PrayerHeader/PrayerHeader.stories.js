import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';
import { PaddedView } from '@apollosproject/ui-kit';

import PrayerHeader from '.';

const headerData = {
  imageSource: {
    uri: 'https://fillmurray.com/400/600',
  },
  name: 'Bill',
  source: 'Anderson',
};

storiesOf('prayer/PrayerHeader', module).add('Anonymous', () => (
  <PaddedView>
    <PrayerHeader
      anonymous
      avatarSize={'medium'}
      avatarSource={headerData.imageSource}
      name={`Pray For ${headerData.name}`}
      source={headerData.source}
    />
  </PaddedView>
));

storiesOf('prayer/PrayerHeader', module).add('Not Anonymous', () => (
  <PaddedView>
    <PrayerHeader
      avatarSize={'medium'}
      avatarSource={headerData.imageSource}
      name={`Pray For ${headerData.name}`}
      source={headerData.source}
    />
  </PaddedView>
));
