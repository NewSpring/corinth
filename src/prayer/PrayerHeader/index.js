import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import { Avatar, H3, H6, styled } from '@apollosproject/ui-kit';

const AvatarView = styled(({ theme }) => ({
  marginBottom: theme.sizing.baseUnit,
}))(View);

const GreyH6 = styled(({ theme }) => ({
  marginTop: theme.sizing.baseUnit * 0.5,
  color: theme.colors.text.tertiary,
}))(H6);

const PrayerHeader = (props) => (
  <View>
    <AvatarView>
      <Avatar
        isLoading={props.loading}
        source={props.avatarSource}
        size={props.avatarSize}
      />
    </AvatarView>
    <H3>{props.title}</H3>
    {props.source ? <GreyH6>{props.source}</GreyH6> : null}
  </View>
);

PrayerHeader.propTypes = {
  avatarSize: PropTypes.string,
  avatarSource: PropTypes.shape({ uri: PropTypes.string }),
  title: PropTypes.string,
  source: PropTypes.string,
  loading: PropTypes.bool,
};

PrayerHeader.defaultProps = {
  avatarSize: 'small',
  title: 'Request',
  source: '',
};

export default PrayerHeader;
