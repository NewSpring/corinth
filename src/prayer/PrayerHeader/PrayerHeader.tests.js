import React from 'react';
import renderer from 'react-test-renderer';

import Providers from 'newspringchurchapp/src/Providers';

import PrayerHeader from '.';

const headerData = {
  imageSource: {
    uri: 'https://fillmurray.com/400/600',
  },
  name: 'Bill',
  source: 'Anderson',
};

describe('the PrayerHeader component', () => {
  it('renders the prayer header', () => {
    const tree = renderer.create(
      <Providers>
        <PrayerHeader
          avatarSize={'medium'}
          avatarSource={headerData.imageSource}
          name={`Pray For ${headerData.name}`}
          source={headerData.source}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders an anonymous prayer header', () => {
    const tree = renderer.create(
      <Providers>
        <PrayerHeader
          anonymous
          avatarSize={'medium'}
          avatarSource={headerData.imageSource}
          name={`Pray For ${headerData.name}`}
          source={headerData.source}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
