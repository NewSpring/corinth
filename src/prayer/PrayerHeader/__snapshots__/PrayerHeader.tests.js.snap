// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the PrayerHeader component renders an anonymous prayer header 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "flex-end",
          "width": 80,
        }
      }
      themeSize={80}
    >
      <View
        animate={null}
        customAnimate={null}
        onReady={false}
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#dddddd",
            "borderRadius": 40,
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      >
        <Image
          onLoad={[Function]}
          size="medium"
          source={
            Array [
              Object {
                "cache": "force-cache",
                "uri": "https://fillmurray.com/400/600",
              },
            ]
          }
          style={
            Object {
              "backgroundColor": "#dddddd",
              "borderRadius": 40,
              "bottom": 0,
              "height": "100%",
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
          themeSize={80}
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Black",
        "fontSize": 24,
        "lineHeight": 27.6,
      }
    }
  >
    Request
  </Text>
  <Text
    style={
      Object {
        "color": "#858585",
        "fontFamily": "Colfax-Bold",
        "fontSize": 12,
        "lineHeight": 17.28,
        "marginTop": 8,
      }
    }
  >
    Anderson
  </Text>
</View>
`;

exports[`the PrayerHeader component renders the prayer header 1`] = `
<View>
  <View
    style={
      Object {
        "marginBottom": 16,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "flex-end",
          "width": 80,
        }
      }
      themeSize={80}
    >
      <View
        animate={null}
        customAnimate={null}
        onReady={false}
        style={
          Object {
            "aspectRatio": 1,
            "backgroundColor": "#dddddd",
            "borderRadius": 40,
            "bottom": 0,
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      >
        <Image
          onLoad={[Function]}
          size="medium"
          source={
            Array [
              Object {
                "cache": "force-cache",
                "uri": "https://fillmurray.com/400/600",
              },
            ]
          }
          style={
            Object {
              "backgroundColor": "#dddddd",
              "borderRadius": 40,
              "bottom": 0,
              "height": "100%",
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "100%",
            }
          }
          themeSize={80}
        />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#303030",
        "fontFamily": "Colfax-Black",
        "fontSize": 24,
        "lineHeight": 27.6,
      }
    }
  >
    Request
  </Text>
  <Text
    style={
      Object {
        "color": "#858585",
        "fontFamily": "Colfax-Bold",
        "fontSize": 12,
        "lineHeight": 17.28,
        "marginTop": 8,
      }
    }
  >
    Anderson
  </Text>
</View>
`;
